### ISSUE TITLE
Notify role="dialog" requires a label

### ISSUE THREAD

**Issue by kirsty-hames** [adaptlearning/adapt-contrib-core#483]
### Subject of the issue
Notify uses `dialog` role to help assistive technologies identify the dialog's content as being grouped and separated from the rest of the page content. Adding `role="dialog"` alone is not sufficient to make a dialog accessible and must be properly labeled. See [MDN ARIA: dialog role description for reference](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role#description).

Both [NotifyPopupView](https://github.com/adaptlearning/adapt-contrib-core/blob/6657b362f59cccbb71e49b0e306227fd7c25ddce/js/views/notifyPopupView.js#L17) and [NotifyPushView](https://github.com/adaptlearning/adapt-contrib-core/blob/6657b362f59cccbb71e49b0e306227fd7c25ddce/js/views/notifyPushView.js#L17) use `aria-labelledby` and reference the Notify title to label the dialog however title properties aren't set as required and are sometimes removed in favour of having body text only.

In the instance no title is set, the `dialog` role isn't identified and the user is unaware the content is grouped/separate from the main page content.

Notify is utilised by core and plugins. See below (and please add any I've missed):

- Core question model
- Bookmarking
- LanguagePicker
- Hotgraphic
- Narrative
- Hotgrid
- Visua11y

#### Example using MCQ feedback

When `_feedback title` is set, `.notify__title` is used to label the popup `dialog`. Which reads _"Feedback dialog"_.

<img width="1133" alt="notify_with_title" src="https://github.com/adaptlearning/adapt-contrib-core/assets/7045330/704116ee-1cb7-4f7d-a3c4-6addbcbf46f2">

When no `_feedback title` is set,`.notify__title` is still referenced by `aria-labelledby` but the element doesn't exist so the popup `dialog` has no label. _"dialog"_ isn't announced.

<img width="1139" alt="notify_without_title" src="https://github.com/adaptlearning/adapt-contrib-core/assets/7045330/03da56b3-329b-46c9-ac77-aaa6e23a38e8">

### Expected behaviour
I think most content authors won't be aware of the accessibility implications by not setting a title so either highlighting this in the instruction/help text, setting titles as required or providing a fallback should prevent this.

**Comment by kirsty-hames** [comment]
**Core question model**
#484 sets a default value _"Feedback"_ for `_globals._accessibility.altFeedbackTitle`. This provides a global feedback title in the instance a component level `_feedback.title` (or `_feedback.altTitle` isn't set).

**Bookmarking**
A [`default`](https://github.com/adaptlearning/adapt-contrib-bookmarking/blob/06ce08a51fcd187b948f18975b42c9144b067a00/properties.schema#L63) value is already set for `_bookmarking.title`.
I'd suggest replacing `default` `"Bookmarking"` with something more meaningful such as`"Continue?"`.

**LanguagePicker**
`warningTitle` sets the dialog title and is already set as [`required`](https://github.com/adaptlearning/adapt-contrib-languagePicker/blob/be74bb8381bcf38f4561fd6ce892105fcdfa2c3d/properties.schema#L234) - no further action.

**Visua11y**
A [`default`](https://github.com/cgkineo/adapt-visua11y/blob/63a38f881db16e882122d09c87929abb7351e66b/properties.schema#L98) value is already set for `_visua11y.title`.

**Hotgraphic, Narrative and Hotgrid**
The dialog title is set by `_items[].title`. It's difficult to set default values as component titles will be content specific. I'd suggest making `_items[].title` required. Alternatively and less favourable, create a generic `aria-label` in the instance a title doesn't exist e.g. _"Item 1 of 3"_.

Overall, I think if a property is required to enable accessibility, then those properties should be set as required.

Please add any comments or suggestions below. I'll wait for feedback before raising issues/PRs for the suggestions above.

**Comment by kirsty-hames** [comment]
Closing as all plugins listed now have a suitable fallback or a required property to provide an accessible dialog label.
